crates/sui-core/src/authority/authority_store.rs

```diff
@@ -45,6 +45,13 @@ pub struct SuiDataStore<const ALL_OBJ_VER: bool, const USE_LOCKS: bool, S> {
     ///
     /// Note that while this map can store all versions of an object, in practice it only stores
     /// the most recent version.
+    ///
+    /// Consistency requirements:
+    /// - Used to guard against regenesis (which asserts that the objects map is empty)
+    /// - Must be updated before new object locks are created so that future transactions can access
+    ///   the newly-created versions
+    /// - Read-after-write consistency for the client (after confirmation_transaction returns, this
```

Comment:
@gdanezis if this is eventual, it means that the client will not have a guarantee that they can exist right away.   But saying 
"without retries" implies a much tighter guarantee.  What do we really want to say here?   Some variant of "most likely right away" or "really soon" still implies that, sometimes it might not be soon.

Reply:
I recognize that @gdanezis is pointing out what is required from the sui-protocol-level POV. I think from the point of view of our specific implementation we should guarantee read-after-write consistency to clients with respect to confirmation_transaction. It simplifies so much about coding against our system.

Additionally, I don't see what insisting on eventual consistency at the level of a single key-value store would buy us. The ability to squeeze out a few more milliseconds of latency in some future where a single validator writes its data to a multi-AZ distributed store?